### DISCUSSION TITLE
Prisma, Error: P1014 The underlying table for model `(not available)` does not exist.

### DISCUSSION THREAD

**Discussion by JDuarthe** [prisma/prisma#5591] [Q&A]
Hi everyone, I'm new with prisma and I been doing the getting started example https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch-typescript-postgres

And got this error while doing the migration
`Error: P1014`
`The underlying table for model (not available) does not exist.`

someone know what i'm doing wrong?

prisma version: 2.16.1

**Comment by mandaputtra** [comment]
You're maybe missing the model that you want to migrate. Can you share some code?

**Reply by JDuarthe** [reply]
this is my schema.prisma, 

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Post {
  id        Int      @default(autoincrement()) @id
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String
  content   String?
  published Boolean  @default(false)
  author    User     @relation(fields: [authorId], references: [id])
  authorId  Int
}

model Profile {
  id     Int     @default(autoincrement()) @id
  bio    String?
  user   User    @relation(fields: [userId], references: [id])
  userId Int     @unique
}

model User {
  id      Int      @default(autoincrement()) @id
  email   String   @unique
  name    String?
  posts   Post[]
  profile Profile?
}
```

this is the response in the terminal

PS D:\practicas\barcos\server> npx prisma migrate dev --name init --preview-feature
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": PostgreSQL database "barcos", schema "public" at "localhost:5432"

Error: P1014

The underlying table for model `(not available)` does not exist.

**Reply by mandaputtra** [reply]
Try change the user model like these, it maybe syntax error since I don't see any usage of `Profile?` in docs.

```
model User {
  id      Int      @default(autoincrement()) @id
  email   String   @unique
  name   String?
  posts    Post[]
  profile  Profile[]
}
```

**Reply by JDuarthe** [reply]
I found the error, just needed to run npx prisma migrate reset --preview-feature and did the migration after that without problem.

**Reply by snake-py** [reply]
You just saved my sanity.

**Reply by shamxeed** [reply]
@JDuarthe thanks! worked for me too!!

**Reply by iamthewalkingkode** [reply]
The above commands resets your DB. All data will be lost. Use at your own risk.

**Reply by The-Kevin** [reply]
Take care about this command, bro
this redefine the database, please avoid this command in production environment

**Reply by kiwavi** [reply]
Had a similar issue. The problem was that I had not committed one migration file that created the table but I had committed a migration file that was to edit that table. So it was trying to edit a table that was not created, hence the error. Fixed by staging the migration file for table creation.

**Reply by tommy457** [reply]
for me i was trying to create a trigger and run it as a migration file
`AFTER INSERT OR DELETE ON "Comment" `
table names are case sensitive and put table name in quotes

**Comment by Glasgesicht** [comment]
This very issue popped up for me after changing 
`textField            String? @db.NVarChar(Max)`
to 
`textField            String? @db.NVarChar(255)`
using MSSQL as database

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
